src/bbs-plus-presentation.js

```diff
@@ -0,0 +1,81 @@
+import { BBSPlusPublicKeyG2, initializeWasm } from '@docknetwork/crypto-wasm-ts';
+import b58 from 'bs58';
+import {
+  PresentationBuilder,
+  Credential,
+} from '@docknetwork/crypto-wasm-ts/lib/anonymous-credentials';
+import { Presentation } from '@docknetwork/crypto-wasm-ts/lib/anonymous-credentials/presentation';
+import { ensureArray } from './utils/type-helpers';
+
+import Bls12381BBSSignatureDock2022 from './utils/vc/crypto/Bls12381BBSSignatureDock2022';
+import CustomLinkedDataSignature from './utils/vc/crypto/custom-linkeddatasignature';
+
+export default class BbsPlusPresentation {
+  /**
+   * Create a new BbsPlusPresentation instance.
+   * @constructor
+   */
+  constructor() {
+    this.presBuilder = new PresentationBuilder();
+  }
+
+  /**
+   * Species attributes to be revealed in the credential
+   * @param {number} credentialIndex
+   * @param {Array.<string>} attributes
+   */
+  addAttributeToReveal(credentialIndex, attributes = []) {
+    ensureArray(attributes);
+    this.presBuilder.markAttributesRevealed(credentialIndex, new Set(attributes));
+  }
+
+  /**
+   * Creates a presentation from the added credentials
+   * @returns {object}
+   */
+  createPresentation() {
+    const pres = this.presBuilder.finalize();
+    return pres.toJSON();
+  }
+
+  /**
+   * Add jsonld credentials to be presented.
+   * @param credentialLD
+   * @param publicKey
+   * @returns {Promise<number>}
+   */
+  async addCredentialsToPresent(credentialLD, publicKey) {
+    await initializeWasm();
+    const json = typeof credentialLD === 'string' ? JSON.parse(credentialLD) : credentialLD;
+
+    const pkRaw = b58.decode(publicKey);
+    const pk = new BBSPlusPublicKeyG2(pkRaw);
+
+    const [credential] = await Bls12381BBSSignatureDock2022.convertCredential({
+      document: json,
+    });
+
+    return this.presBuilder.addCredential(Credential.fromJSON(credential, CustomLinkedDataSignature.fromJsigProofValue(credentialLD.proof.proofValue)), pk);
+  }
+
+  /**
+   *
+   * @param presentationLD
+   * @param {Array.<string>} publicKeys
+   * @returns {boolean}
+   */
+  verifyPresentation(presentationLD, publicKeys) {
```

Comment:
also @osinakayah this method shouldnt be in this class because it doesnt use the presentation builder, better to just export it as a `verifyPresentation` function. it should also be updated to accept a documentLoader parameter and get the issuer public keys from each credential's revealed `verificationMethod` field

Reply:
alternative you could modify this method https://github.com/docknetwork/sdk/blob/master/src/utils/vc/presentations.js#L104 to detect a BBS+ presentation and apply above feedback, this way we can use the same `verifyPresentation(...)` api. if not id personally put a new method here called `verifyBBSPresentation` just to keep things organised a bit better

Reply:
What if the issuerDID or verificationMethod field isn't revealed, is there another way to get the issuer public keys?

Reply:
nope, thats why you must enforce its revealed in this wrapper code (eg when a credential is added, call revealField(proof.verificationMethod) which will give the key ID - pass that to documentLoader and it will load the key JSON object from which you can get the base58 encoded pubic key